pyarn/lockfile.py

```diff
@@ -142,4 +142,7 @@ def _quote_key_if_needed(key):
 
 
 def _needs_quoting(s):
+    if s.startswith('true') or s.startswith('false'):
+        # If a string starts with a boolean, it must be quoted no matter what
```

Comment:
Looks great to me! Thank you for your explanation. 

I am curious, can we elaborate on the reason for quoting in the comment more? To point that it is a special token for yarn lexer even it is a part of a string. 

If it is not needed, LGTM

Reply:
I added something, does that explain it better?

Reply:
Yes, thank you!